src/sql/parts/connection/connectionDialog/connectionWidget.ts

```diff
@@ -462,18 +530,15 @@ export class ConnectionWidget {
 			this._passwordInputBox.focus();
 			isFocused = true;
 		}
-		let validateDatabaseName = this._databaseNameInputBox.validate();
-		if (!validateDatabaseName && !isFocused) {
-			this._databaseNameInputBox.focus();
-		}
-		return validateServerName && validateUserName && validatePassword && validateDatabaseName;
+
+		return validateServerName && validateUserName && validatePassword;
 	}
 
 	public connect(model: IConnectionProfile): boolean {
 		let validInputs = this.validateInputs();
 		if (validInputs) {
 			model.serverName = this.serverName;
-			model.databaseName = this.databaseName;
+			model.databaseName = this.databaseName === this._defaultDatabaseName ? 'master' : this.databaseName;
```

Comment:
>this._defaultDatabaseName ? 'master' [](start = 46, length = 36)

the default DB won't always be master, so I think this is incorrect.  Can we go back to the previous behavior where a blank value means default and we let the server fill in what the default DB is once it connects?